### ISSUE TITLE
How "validation" row_role should be used during prediction?

### ISSUE THREAD

**Issue by Fred-Wu** [mlr-org/mlr3#767]
I am new to mlr3verse, and I think the workflow of mlr3 is easy to understand and implement. Thanks for creating this. 

One of the immediate question to me during learning is how "validation" row_role should be used when doing prediction? The example sets row_ids for training and test sets separately, and manually set them in leaner$train, and learner$predict. 

The book says if the target is missing, those rows will automatically be set as validation set. Should this always be done before creating the task?

**Comment by mllg** [comment]
> One of the immediate question to me during learning is how "validation" row_role should be used when doing prediction? The example sets row_ids for training and test sets separately, and manually set them in leaner$train, and learner$predict.
> 
> The book says if the target is missing, those rows will automatically be set as validation set. Should this always be done before creating the task?

This is still a matter we have to discuss internally. 
Right now, rows with role "validation" are just kept in the task, but not used for resampling. They can be used in performance measures, though. 

I've put this topic onto the agenda for our next internal meeting, and will update the book accordingly afterwards.

**Comment by Fred-Wu** [comment]
I am still trying to understand how the now-renamed `holdout` dataset in a task `row_roles` can be used in the prediction stage. I think it is related to the concept of the package. 

For the following codes, is the `row_roles`'s holdout for the resampling purpose or is it kept as a final "test" dataset?

```
task <- as_task_classif(data_store, target = "disposition", positive = "Admit")
i_test <- sample(task$nrow, 56000)
i_train <- task$row_ids[-i_test]
task$set_row_roles(i_test, add_to = "holdout")

> length(i_test)
[1] 56000
> length(i_train)
[1] 504486
> task$nrow
[1] 560486
```

However, setting the `holdout` does not remove that portion of data from the `use` unless remove them explicitly with
```
task2 <- task$clone()
task2$set_row_roles(i_test, add_to = "holdout", remove_from = 'use')

> task2$nrow
[1] 504486

```

If do not specify `remove_from`, those data would be used in the training. 
If do remove them from `use`, then I have no idea what's the purpose of `holdout` in a task and how I can use it.

**Comment by mllg** [comment]
You can just assign a new role via 
```r
task$set_row_roles(i_test, roles = "holdout")
```

https://mlr3.mlr-org.com/reference/Task.html#method-set-row-roles-

**Comment by Fred-Wu** [comment]
> You can just assign a new role via
> 
> ```r
> task$set_row_roles(i_test, roles = "holdout")
> ```
> 
> https://mlr3.mlr-org.com/reference/Task.html#method-set-row-roles-

Thanks for letting me know. 

Then, will this "holdout" data be served as the final "test" data as to apply on `predict_newdata()`, or is it kept for now without any purposes as mentioned in the earlier reply?

Or will it be used in 
```
lrns("classif.rpart", predict_type = "prob", predict_sets = c("train", "test", "holdout"))
```

I think my question comes from what these are called in a model development process from the package perspective. 

1. training, validation, testing/holdout (data kept outside the development process)
Or
2. training, testing, holdout.
